hal/src/thumbv7em/clock/v2/xosc32k.rs

```diff
@@ -151,109 +153,104 @@ impl Sealed for CrystalMode {}
 // Xosc32k
 //==============================================================================
 
-pub struct Xosc32k<M>
+pub struct Xosc32k<M, X, Y>
 where
     M: Mode,
+    X: Output32k,
+    Y: Output1k,
```

Comment:
Wasn't this previously a run-time parameter, not compile-time? What's the motivation for this change? Maybe I just don't understand the `Xosc32k` very well?